File: common/src/main/java/com/rpmtw/rpmtw_platform_mod/gui/ConfigScreen.java
package com.rpmtw.rpmtw_platform_mod.gui;

import me.shedaniel.autoconfig.ConfigData;
import me.shedaniel.autoconfig.annotation.Config;
import me.shedaniel.autoconfig.annotation.ConfigEntry;
import me.shedaniel.cloth.clothconfig.shadowed.blue.endless.jankson.Comment;
import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;

@Config(name = "rpmtw_platform_mod")
@Environment(EnvType.CLIENT)
public class ConfigScreen implements ConfigData {

    @ConfigEntry.Category("translate")
    @ConfigEntry.Gui.TransitiveObject
    public Translate translate = new Translate();

    public static class Translate {
        @Comment("啟動遊戲時自動切換遊戲語言至 RPMTranslator 支援的語言")
        public boolean autoToggleLanguage = true;
    }
}

File: common/src/main/kotlin/com/rpmtw/rpmtw_platform_mod/gui/ConfigScreen.kt
package com.rpmtw.rpmtw_platform_mod.gui

import me.shedaniel.autoconfig.ConfigData
import me.shedaniel.autoconfig.annotation.Config
import me.shedaniel.autoconfig.annotation.ConfigEntry
import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment

@Config(name = "rpmtw_platform_mod")
@Environment(EnvType.CLIENT)
class ConfigScreen : ConfigData {
    @JvmField
    @ConfigEntry.Category("base")
    @ConfigEntry.Gui.TransitiveObject
    var base = Base()

    @JvmField
    @ConfigEntry.Category("translate")
    @ConfigEntry.Gui.TransitiveObject
    var translate = Translate()

    @JvmField
    @ConfigEntry.Category("cosmicChat")
    @ConfigEntry.Gui.TransitiveObject
    var cosmicChat = CosmicChat()


    class Base {
        @JvmField
        @ConfigEntry.Gui.Excluded
        var rpmtwAuthToken: String? = null
    }

    class Translate {
        @JvmField
        @ConfigEntry.Gui.Tooltip(count = 1)
        @ConfigEntry.Gui.RequiresRestart
        var autoToggleLanguage = true
    }

    class CosmicChat {
        @JvmField
        @ConfigEntry.Gui.Tooltip(count = 1)
        @ConfigEntry.Gui.RequiresRestart
        var enable = true

        @JvmField
        var enableReceiveMessage = true

        @JvmField
        @ConfigEntry.Gui.Tooltip(count = 1)
        var enableButton = true

        @JvmField
        @ConfigEntry.Gui.Excluded
        var eula = false

        @JvmField
        @ConfigEntry.Gui.Tooltip(count = 1)
        var nickname: String? = null

        @JvmField
        @ConfigEntry.Gui.Tooltip(count = 1)
        @ConfigEntry.Gui.RequiresRestart
        @ConfigEntry.Gui.EnumHandler(option = ConfigEntry.Gui.EnumHandler.EnumDisplayOption.BUTTON)
        var accountType: CosmicChatAccountType = CosmicChatAccountType.RPMTW
    }
}

enum class CosmicChatAccountType {
    MINECRAFT,
    RPMTW
}